setup.py

```diff
@@ -70,9 +63,6 @@ def get_petsc_dir():
                         libraries=libs,
                         extra_link_args=link_args,
                         cython_compile_time_env=cython_compile_time_env) for (ext, libs) in cythonfiles]
-if 'CC' not in env:
```

Comment:
Why does this have to go? Maybe you want to guard it with `if config["options"].get("install_mode", "default") == "default":`?

Reply:
Note that in case you do want to guard it with that you also need to `del config` later.

Reply:
This is already set in `firedrake-install` ([here](https://github.com/firedrakeproject/firedrake/blob/master/scripts/firedrake-install#L26)). I think that it is redundant.